browser/resources/settings/brave_default_extensions_page/brave_default_extensions_page.js

```diff
@@ -40,6 +41,16 @@ Polymer({
     this.browserProxy_.setIPFSCompanionEnabled(this.$.ipfsCompanionEnabled.checked);
   },
 
+  restartBrowser_: function() {
+    this.browserProxy_.restartBrowser();
+  },
+
+  onMediaRouterEnabledChange_: function() {
+    this.browserProxy_.setMediaRouterEnabled(this.$.mediaRouterEnabled.checked);
+    this.$.restartButton.onclick = this.browserProxy_.restartBrowser
+    this.$.needsRestart.hidden = false;
```

Comment:
Doesn't this mean that if this page is re-visited when the browser wasn't restarted, the 'restart' label doesn't show? On `brave://flags`, it persists when the page is opened again. Perhaps we can store a browser in-memory 'needs-restart' value for media router?

Reply:
Will investigate, thanks!